public/electron.js

```diff
@@ -163,21 +157,28 @@ ipc.on('trigger', (event, args) => {
 let stream = false
 let fileName = ''
 let filePath = ''
-let patientID = ''
+let participantID = ''
+let studyID = ''
 let images = []
 let startTrial = -1
 
 // Read version file (git sha and branch)
 var git = JSON.parse(fs.readFileSync(path.resolve(__dirname, 'config/version.json')));
 
+// Get Participant Id and Study Id from environment
+ipc.on('syncCredentials', (event) => {
+  event.returnValue = {envParticipantId: process.env.REACT_APP_PARTICIPANT_ID, envStudyId: process.env.REACT_APP_STUDY_ID}
+})
+
 // listener for new data
 ipc.on('data', (event, args) => {
 
-  // initialize file - we got a patinet_id to save the data to
-  if (args.patient_id && fileName === '') {
+  // initialize file - we got a participant_id to save the data to
+  if (args.participant_id && args.study_id && fileName === '') {
     const dir = app.getPath('userData')
-    patientID = args.patient_id
-    fileName = `pid_${patientID}_${Date.now()}.json`
+    participantID = args.participant_id
+    studyID = args.study_id
+    fileName = `pid_${participantID}_${Date.now()}.json`
     filePath = path.resolve(dir, fileName)
```

Comment:
`studyId` isn't used anywhere - should it be?  is the `filePath` resolution correct?

Reply:
Yeah it should be in the path but not in the name itself